process/handlers.lua

```diff
@@ -326,14 +341,23 @@ function handlers.evaluate(msg, env)
   local handled = false
   assert(type(msg) == 'table', 'msg is not valid')
   assert(type(env) == 'table', 'env is not valid')
-  
+
   for _, o in ipairs(handlers.list) do
     if o.name ~= "_default" then
       local match = utils.matchesSpec(msg, o.pattern)
       if not (type(match) == 'number' or type(match) == 'string' or type(match) == 'boolean') then
         error("Pattern result is not valid, it MUST be string, number, or boolean")
       end
-      
+
+      -- ensure the handler hasn't timed out yet
+      if o.timeout then
+        -- remove handler if it timed out
+        if (o.timeout.type == 'milliseconds' and o.timeout.value < msg.Timestamp) or (o.timeout.type == 'blocks' and o.timeout.value < msg["Block-Height"]) then
```

Comment:
I think the having the handler still run at the deadline and only get removed after is the most straightforward implementation/solution, but I'm happy to change it to:
```lua
if (o.timeout.type == 'milliseconds' and o.timeout.value <= msg.Timestamp) or (o.timeout.type == 'blocks' and o.timeout.value <= msg["Block-Height"]) then
```